File: www/cgi/perl/submit.pl
#!/usr/bin/perl

use warnings;
use strict;
use POSIX;
use URI::Escape;
use Encode;
use CGI qw(:standard);
use File::Basename;

my $doc_root = $ENV{'DOCUMENT_ROOT'};
my $success = open LOG,'>', "/tmp/".basename(__FILE__).'_'.getgid().'.txt';
if (!$success) {
	exit 1;
}

print LOG __LINE__,"==\n";

my $server_addr=$ENV{'SERVER_ADDR'}; 
my $server_port=$ENV{'SERVER_PORT'}; 
my $http_referer=$ENV{'HTTP_REFERER'};
my $method=$ENV{'REQUEST_METHOD'};

if(!($method =~ /POST/)) {
	printf LOG "GET\n";
#	print "Content-type: text/html\r\n";
#	print "\r\n";
#	print "error method";
#	exit;
}
else
{
	printf LOG "POST\n";
}

#if(!$server_addr){
#	$server_addr = "127.0.0.1";
#}

my $firstname=param('firstname');
if(not defined $firstname) {
	$firstname="?";
}

my $lastname=param('lastname');
if(not defined $lastname) {
	$lastname="?";
}

my $time = localtime;

printf LOG "firstname=%s\nlastname=%s\n",$firstname,$lastname;
# 
print "Content-type: text/html; charset=UTF-8\n\n";
print <<END_OF_PAGE;
<!DOCTYPE html>
<html>
 <head>
  <title>Test Perl</title>
 </head>
 <body leftmargin="100" rightmargin="100" bgcolor="#00bf0f">
  <h3>Здравствуйте, $firstname $lastname!</h3>
  <form action="submit.pl" method="$method">
    <p>Just type in your name (and click Submit) to enter the contest:<br>
       First name: <input type="text" name="firstname" value=""><br>
       Last name:  <input type="text" name="lastname" value=""><br>
       <input type="submit" value="Submit">
    </p>
  </form>
  <hr>
  $time
 </body>
</html>
END_OF_PAGE

File: www/cgi/python/submit.py
#!/usr/bin/env python3

import os, datetime 
import cgi
import codecs,sys

sys.stdout = codecs.getwriter("utf-8")(sys.stdout.detach(), errors='replace')
form = cgi.FieldStorage()
text1 = form.getfirst("TEXT_1", "?")
text2 = form.getfirst("TEXT_2", "?")
try:
	meth = os.environ['REQUEST_METHOD']
except:
	meth = 'get'

print("Content-Type: text/html\n")

print("""<!DOCTYPE HTML>
<html>
  <head>
   <meta charset="utf-8">
   <title>Test Form</title>
  </head>
  <body>
   <meta charset=\"utf-8\">""")
print("   <h3>","Hello ",text1,' ', text2, "!</h3>")
print("   <p>   len=%d</p>"%len(text1))
print("""   <form action=\"/cgi-bin/python/submit.py\" method=\"%s\" >
    <input type=\"text\" name=\"TEXT_1\" value=\"\">
    <input type=\"text\" name=\"TEXT_2\" value=\"\"><br>
    <input type=\"submit\">
   </form>""" %(meth))

now = datetime.datetime.utcnow()
print("""   <hr>
   %s
  </body>
</html>""" %now.strftime("%a, %d %b %Y %H:%M:%S GMT"))
